examples_utils/benchmarks/run_benchmarks.py

```diff
@@ -240,13 +252,19 @@ def run_benchmark_variant(
         subprocess.check_output([sys.executable, "-m", "pip", "install", "-r", str(reqs)])
 
     logger.info(f"Start test: {start_time}")
-    stdout, stderr, exitcode = run_and_monitor_progress(
-        cmd,
-        listener,
-        args.timeout,
-        cwd=cwd,
-        env=env,
-    )
+    attempt_run = True
+    while attempt_run:
+        stdout, stderr, exitcode = run_and_monitor_progress(
+            cmd,
+            listener,
+            args.timeout,
+            cwd=cwd,
+            env=env,
+        )
+        attempt_run = reattempt_run(benchmark_dict, stdout, stderr, exitcode)
```

Comment:
I feel calling "reattempt_run" here gives the impression that the run is being re-attempted. That being said, I cant think of anything better 😭 

I get that you shouldnt use "check_if_true" kinda names for bool output functions but this feels like a different case... I cant quite put it down as a thought though. 

anyway, LGTM.

Reply:
I agree, what about: 

```suggestion
        need_to_run = should_reattempt_benchmark(benchmark_dict, stdout, stderr, exitcode)
```

Reply:
That looks good too